### PR TITLE
Fixes for package.py

### PR DESCRIPTION
**Pull request by ohehir**
This PR was created to remove duplicated code in package.py that already exists in eodatasets3. This PR also contains a minor modification where platform has been added to utils.get_collection_number() in the non_standard_packaging function in package.py.

### CODE REVIEW

**Comment by ohehir** [comment]
Hi @jeremyh, can you please check/approve. Many thanks, Aaron

**Comment by jeremyh** [comment]
We probably should add an eodataset3 minimum version dependnecy to `setup.py`, since this seems to be tied to it.

**Comment by ohehir** [comment]
OK, I'll add "eodatasets>0.19.2" to setup.py

**Comment by ohehir** [comment]
Hi @jeremyh, I've added the eodatasets requirement to setup.py
Please check/approve/merge when ready.
Aaron

File: setup.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
         "shapely",
         "structlog",
         "checksumdir",
-        "eodatasets",
+        "eodatasets>=0.19.2",
```
Comment (jeremyh) [inline setup.py:28]: `eodatasets` and `eodatasets3` are different libraries. I think we need both here until the old one is phased out.

Comment (ohehir) [reply setup.py]: ok, I've changed setup.py to:
`        "eodatasets",
        "eodatasets3>=0.19.2",`

**Review by jeremyh — APPROVED** [review]

**Pull request merged**